filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilder.scala

```diff
@@ -68,9 +69,16 @@ class BlobPathBuilder(credential: AzureSasCredential, container: String, endpoin
     validateBlobPath(string, container, endpoint) match {
       case ValidBlobPath(path) =>
         Try {
-          val fileSystem = FileSystems.newFileSystem(new URI("azb://?endpoint=" + endpoint), fileSystemConfig.asJava)
-          val blobStoragePath = fileSystem.getPath(path)
-          BlobPath(blobStoragePath, endpoint, container)
+          val fileSystem = Try {
+            FileSystems.getFileSystem(new URI("azb://?endpoint=" + endpoint))
+          } recover {
+            // If no filesystem already exists, this will create a new connection, with the provided configs
+            case _: FileSystemNotFoundException => FileSystems.newFileSystem(new URI("azb://?endpoint=" + endpoint), fileSystemConfig.asJava)
+          }
+
+          val blobStoragePath = fileSystem.map(_.getPath(path))
+          // If there is a remaining issue reaching the filesystem, this will rethrow the exception
+          blobStoragePath.map(BlobPath(_, endpoint, container)).get
```

Comment:
Generally in Scala we don't want unprotected `.get` calls resulting in exceptions, even when we expect and properly handle those exceptions. This is partially style and partially performance - my understanding is that throwing exceptions up and down deep stacks is very expensive, so it's good to get out of the habit.

I think there's a different way to structure this whole block so it returns the InvalidFilePath objects directly rather than throwing and creating them in recovery, we can maybe workshop it tomorrow.

Reply:
Thanks for the feedback here, I figured this wasn't a great way to go, but more generally I wasn't able to find great advice on handling nested try situations overall. I realized now looking at it again, I think the nested try is totally unnecessary 😅 Sounds good!